diff -r 0c011dc6dbc4 script/ethercatctl.in
--- a/script/ethercatctl.in	Fri Jun 14 12:42:52 2019 +0200
+++ b/script/ethercatctl.in	Sun Dec 01 01:13:29 2019 +0100
@@ -108,6 +108,16 @@
             continue # ec_* module not found
         fi
 
+	if [ ${MODULE} == "ccat_netdev" ]; then
+	    if ! ${LSMOD} | grep "^${ECMODULE} " > /dev/null ; then
+		# try to unload Beckhoff ccat_netdev
+		if ! ${RMMOD} ${MODULE}; then
+                    ${RMMOD} ${LOADED_MODULES}
+                    exit 1
+		fi
+	    fi
+	fi
+	
         if [ ${MODULE} != "generic" -a ${MODULE} != "ccat" ]; then
             # try to unload standard module
             if ${LSMOD} | grep "^${MODULE} " > /dev/null; then
